azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/implementation/KubernetesClusterImpl.java

```diff
@@ -366,4 +356,25 @@ public KubernetesClusterImpl withRBACDisabled() {
         this.inner().withSku(sku);
         return this;
     }
+
+    KubernetesClusterImpl addNewAgentPool(final KubernetesClusterAgentPoolImpl agentPool) {
+        if (!isInCreateMode()) {
+            this.addDependency(
+                    new FunctionalTaskItem() {
+                        @Override
+                        public Observable<Indexable> call(final Context context) {
+                            return manager().inner().agentPools().createOrUpdateAsync(
+                                    resourceGroupName(), name(), agentPool.name(), agentPool.getAgentPoolInner()
+                            ).flatMap(new Func1<AgentPoolInner, Observable<Indexable>>() {
+                                @Override
+                                public Observable<Indexable> call(AgentPoolInner agentPoolInner) {
+                                    return context.voidObservable();
+                                }
+                            });
+                        }
+                    });
+        }
```

Comment:
I think agent pool could update in cluster update, no need for do another PUT as dependency.

Reply:
Copied from track2, your code. Any idea?

Reply:
I'll take a look.

Reply:
After my testing, update needs agent pool exists.